### ISSUE TITLE
NULL used as identifier while generate cpp code for TSQL, so compile failed.

### ISSUE THREAD

**Issue by chencang1980** [antlr/antlr4#2029]
NULL is a keyword in T-SQL, but also NULL is a predefined macro (defined by VC runtime).

antlr4 take NULL directly used as enum identifiers and function names, that conflict with VC runtime macro NULL

**Comment by chencang1980** [comment]
oh, maybe this is a duplicated issue as  #1851

**Comment by KvanTTT** [comment]
Try to add underscore to the end of identifier: `NULL_`.

**Comment by chencang1980** [comment]
thanks @KvanTTT,  manual change do works.
If antlr4 auto change it, it do better

**Comment by KvanTTT** [comment]
Yes, this is very demanded [issue](https://github.com/antlr/antlr4/issues/1851) that nevertheless won't be implemented :disappointed:
